`python310Packages.paperwork-backend` on nixos x86_64:

```
___________________________ TestLibinsane.test_scan ____________________________

self = <tests.docscan.tests_libinsane.TestLibinsane testMethod=test_scan>
args = (), kwds = {'result': <TestCaseFunction test_scan>}

    def __call__(self, *args, **kwds):
>       return self.run(*args, **kwds)

/nix/store/xf1k5k05vg3zn7dfcpfh1qa7ga48hi3m-python3-3.10.6/lib/python3.10/unittest/case.py:650: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/xf1k5k05vg3zn7dfcpfh1qa7ga48hi3m-python3-3.10.6/lib/python3.10/unittest/case.py:575: in run
    self._addSkip(result, self, skip_why)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.docscan.tests_libinsane.TestLibinsane testMethod=test_scan>
result = <TestCaseFunction test_scan>
test_case = <tests.docscan.tests_libinsane.TestLibinsane testMethod=test_scan>
reason = False

    def _addSkip(self, result, test_case, reason):
        addSkip = getattr(result, 'addSkip', None)
        if addSkip is not None:
>           addSkip(test_case, reason)
E           TypeError: Skipped expected string as 'msg' parameter, got 'bool' instead.
E           Perhaps you meant to use a mark?

/nix/store/xf1k5k05vg3zn7dfcpfh1qa7ga48hi3m-python3-3.10.6/lib/python3.10/unittest/case.py:473: TypeError
```

--- a/pkgs/applications/office/paperwork/paperwork-backend.nix
+++ b/pkgs/applications/office/paperwork/paperwork-backend.nix
@@ -1,10 +1,6 @@
 { buildPythonPackage
 , lib
 , fetchFromGitLab
-
-, isPy3k
-, isPyPy
-
 , pyenchant
 , scikit-learn
 , pypillowfight
@@ -20,8 +16,13 @@
 , openpaperwork-core
 , openpaperwork-gtk
 , psutil
-
-, pkgs
+, gtk3
+, poppler_gi
+, gettext
+, which
+, shared-mime-info
+, libreoffice
+, unittestCheckHook
 }:
 
 buildPythonPackage rec {
@@ -30,9 +31,6 @@ buildPythonPackage rec {
 
   sourceRoot = "source/paperwork-backend";
 
-  # Python 2.x is not supported.
-  disabled = !isPy3k && !isPyPy;
-
   patchPhase = ''
     echo 'version = "${version}"' > src/paperwork_backend/_version.py
     chmod a+w -R ..
@@ -40,38 +38,48 @@ buildPythonPackage rec {
   '';
 
   propagatedBuildInputs = [
-    pyenchant
-    scikit-learn
-    pypillowfight
-    pycountry
-    whoosh
-    termcolor
-    python-Levenshtein
+    distro
+    gtk3
     libinsane
+    natsort
+    openpaperwork-core
+    pyenchant
+    pycountry
     pygobject3
     pyocr
-    natsort
-    pkgs.poppler_gi
-    pkgs.gtk3
-    distro
-    openpaperwork-core
+    pypillowfight
+    python-Levenshtein
+    poppler_gi
+    scikit-learn
+    termcolor
+    whoosh
   ];
 
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
+  nativeBuildInputs = [
+    gettext
+    shared-mime-info
+    which
+  ];
 
-  nativeBuildInputs = [ pkgs.gettext pkgs.which pkgs.shared-mime-info ];
   preBuild = ''
     make l10n_compile
   '';
 
-  checkInputs = [ openpaperwork-gtk psutil pkgs.libreoffice ];
+  checkInputs = [
+    libreoffice
+    openpaperwork-gtk
+    psutil
+    unittestCheckHook
+  ];
 
-  meta = {
+  preCheck = ''
+    export HOME=$TMPDIR
+  '';
+
+  meta = with lib; {
     description = "Backend part of Paperwork (Python API, no UI)";
-    homepage = "https://openpaper.work/";
-    license = lib.licenses.gpl3Plus;
-    maintainers = with lib.maintainers; [ aszlig symphorien ];
+    homepage = "https://openpaper.work";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ aszlig symphorien ];
   };
 }

